src/towncrier/build.py

```diff
@@ -213,6 +214,16 @@ def __main(
     # in the template.
     render_title = config.title_format == ""
 
+    # Add format-specific context to the template
+    md_header_level: int | None = None
+    if is_markdown:
+        if config.title_format:
+            m = re.search(r"^#+(?=\s)", config.title_format, re.MULTILINE)
+            lvl = len(m[0]) if m else 0
+        else:  # TODO: derive from template or make configurable?
```

Comment:
What do you think we should do here?

Reply:
For RST this is not possible.

This is why we have the `underlines` configuration.

--------

My hope is we can remove the heading markup from the Python code and let this up to the Jinja2 template rendering.

Users can then implement whatever level they want.

Reply:
For rST everything just works anyway. This stuff is just necessary for Markdown (for rST, `None` is always passed in, which leads to the variable not being defined in the template)

> This is why we have the `underlines` configuration.

If you have an idea how to make that work for Markdown please tell me.

Currently all of this just works as it should (see the changed tests)

Reply:
That doesn’t sound like what this PR does, but maybe I’m misunderstanding.

I think if you want to go in a completely different direction, maybe you should make an alternative PR instead. That would be faster than explaining to me what you want and changing this PR to look like that.

Reply:
> We don't have to use the header_prefix value for RST template.
> We can leave it empty.

Check again, this is already the case. The `header_prefix` variable is only passed to Markdown templates, not rST. Or did you mean something else?

Reply:
True...My bad. I am sorry for all the confusing here.

------

I think that we should have a constant value here, that should make the rendered MD file valid wnen using the default template

If users are using a custom template, they will have to also set a custom `title_format`

This can be documenented ?

Make thanks?

Reply:
> I see that the current tests pass

yeah, I repurposed a rather complicated test that required a custom template to just use the now-functional default template